src/api/profile/profile.api.js

```diff
@@ -0,0 +1,118 @@
+import ElevationProfile from '@/api/profile/ElevationProfile.class'
+import ElevationProfilePoint from '@/api/profile/ElevationProfilePoint.class'
+import ElevationProfileSegment from '@/api/profile/ElevationProfileSegment.class'
+import { API_SERVICE_ALTI_BASE_URL } from '@/config'
+import { toLv95 } from '@/modules/drawing/lib/drawingUtils'
+import { CoordinateSystems, splitIfOutOfLV95Bounds } from '@/utils/coordinateUtils'
+import log from '@/utils/logging'
+import { round } from '@/utils/numberUtils'
+import axios from 'axios'
+
+function parseProfileFromBackendResponse(backendResponse, startingDist) {
+    const points = []
+    backendResponse.forEach((rawData) => {
+        points.push(
+            new ElevationProfilePoint(
+                [rawData.easting, rawData.northing],
+                startingDist + rawData.dist,
+                rawData.alts.COMB
+            )
+        )
+    })
+    return new ElevationProfileSegment(points)
+}
+
+/**
+ * @param {CoordinatesChunk} chunk
+ * @param {[Number, Number] | null} startingPoint
+ * @param {Number} startingDist
+ */
+async function getProfileDataForChunk(chunk, startingPoint, startingDist = 0) {
+    if (chunk.isWithinLV95Bounds) {
+        try {
+            const dataForChunk = await axios({
+                url: `${API_SERVICE_ALTI_BASE_URL}rest/services/profile.json`,
+                method: 'POST',
+                params: {
+                    offset: 0,
+                    sr: CoordinateSystems.LV95.epsgNumber,
+                    distinct_points: true,
+                },
+                data: {
+                    type: 'LineString',
+                    coordinates: chunk.coordinates,
+                },
+            })
+            // checking validity of data from the backend, there is small edge case where we request the backend
+            // because the coordinates are technically in bound of LV95, but our backend doesn't cover the last km
+            // or so of the LV95 bounds, resulting in an empty profile being sent by the backend even though our
+            // coordinates were inbound (hence the dataForChunk.data.length > 2 check)
+            if (dataForChunk && dataForChunk.data && dataForChunk.data.length > 2) {
+                return parseProfileFromBackendResponse(dataForChunk.data, startingDist)
+            } else {
+                log.error('Incorrect/empty response while getting profile', dataForChunk)
+            }
+        } catch (err) {
+            log.error('Error while trying to fetch profile data', err)
+        }
```

Comment:
Would be better to explicitely return a value in case of error, e.g. `null` or an empty `ElevationProfileSegment`

Reply:
Oh I just see that it return an empty segment below. I don't really like method that have a return value in the middle, it would be easier to read and understand to have something like
```javascript
let segment = null
if (chunk.isWithinLV95Bounds) {
  ...
  segment = parseProfileFromBackendResponse()
}

if (!segment) {
   // empty segment in case of error or non lv95 bound
   segment = ...
}

return segment
```

Reply:
And I personally prefer a fail fast (return fast) approach so that we do not have too many indentation, rendering the code harder to grasp 😉 
That's totally a case of taste...